apps/namadillo/src/App/Transfer/TransferModule.tsx

```diff
@@ -306,17 +297,27 @@ export const TransferModule = ({
           return 0;
         }
 
-        const asset1Index = findTokenFeeIndex(source.chain, asset1.asset);
-        const asset2Index = findTokenFeeIndex(source.chain, asset2.asset);
-
-        // No fee assets, so we sort them by the amounts owned by the user
-        if (asset1Index === -1 && asset2Index === -1)
-          return asset1.amount.gt(asset2.amount) ? -1 : 1;
-
-        if (asset1Index === -1) return 1;
-        if (asset2Index === -1) return -1;
+        const bip44Index1 = allUsersAssets.findIndex(
+          (asset) =>
+            asset?.symbol.toLowerCase() === asset1.asset?.symbol.toLowerCase()
+        );
+        const bip44Index2 = allUsersAssets.findIndex(
+          (asset) =>
+            asset?.symbol.toLowerCase() === asset2.asset?.symbol.toLowerCase()
+        );
+
+        // If either asset is in the list, sort based on its position.
+        if (bip44Index1 !== -1 || bip44Index2 !== -1) {
+          // If both assets are in the list, sort by their order in registeredCoinTypes.
+          if (bip44Index1 !== -1 && bip44Index2 !== -1) {
+            return bip44Index1 - bip44Index2;
```

Comment:
if both assets are in the list, we should sort by the amount, because this token list is not necessarily the best for user

Reply:
hmm maybe we should just revert back the original implementation and only replace the `findTokenFeeIndex` by the new `findIndex`, wdyt?